In [8]:
import numpy as np
from mostlyai import MostlyAI

In [2]:
# initialize client
mostly = MostlyAI(api_key='mostly-3e555a61ed4b93cd4bbf4e639800f62a9d616c40edb2595666c4deac3348b71b', base_url='https://app.mostly.ai')

In [3]:
# fetch configuration via API
g = mostly.generators.get('1d9f85fd-277c-40d8-985f-02d3f39c948b')
config = g.config()
config

{'name': 'SK Patient Data',
 'description': None,
 'tables': [{'name': 'SK Patient Data',
   'sourceConnectorId': '2f880c16-adf9-4b89-85a6-29bb9316c741',
   'location': 'mostly-production-mostly-s3-bucket-mostly-app/connectors/2f880c16-adf9-4b89-85a6-29bb9316c741/SK Patient Data/cleaned_sk_data.csv',
   'data': None,
   'modelConfiguration': {'maxSampleSize': 6712,
    'batchSize': None,
    'modelSize': 'M',
    'maxTrainingTime': 10,
    'maxEpochs': 100,
    'maxSequenceWindow': 100,
    'enableFlexibleGeneration': False,
    'valueProtection': False,
    'rareCategoryReplacementMethod': 'CONSTANT'},
   'textModelConfiguration': None,
   'primaryKey': None,
   'foreignKeys': None,
   'columns': [{'name': 'latitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'longitude', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'sex', 'modelEncodingType': 'NUMERIC_DISCRETE'},
    {'name': 'age', 'modelEncodingType': 'NUMERIC_DISCRETE'},
    {'name': 'state', 'modelEncodingType'

In [4]:
# probe the generator for some samples
mostly.probe('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=10)

latitude  longitude  sex  age  state
0 -1.724530   1.601258    1    2      0
1  0.887959  -0.808825    1    6      0
2 -0.974640   1.755753    0    2      0
3  0.852118  -0.737850    1    2      0
4 -0.331314  -0.268020    1    3      0
5 -0.753345   2.010923    1    2      0
6 -1.695323   1.621442    1    2      0
7  0.820392  -0.767012    1    2      0
8 -1.739293   1.666134    0    2      0
9  0.651809  -0.987975    1    4      0

We need 20 synthetic data sets. Original data set size was 6712, so we need $6712 \times 20 = 134240$ samples.

In [6]:
# use generator to create a synthetic dataset
sd = mostly.generate('1d9f85fd-277c-40d8-985f-02d3f39c948b', size=134_240)
sd.data()

Created synthetic dataset ]8;id=162747;https://app.mostly.ai/d/synthetic-datasets/0e69c80f-8bc4-4653-92c8-740ace235996\0e69c80f-8bc4-4653-92c8-740ace235996]8;;\ with generator ]8;id=420694;https://app.mostly.ai/d/generators/1d9f85fd-277c-40d8-985f-02d3f39c948b\1d9f85fd-277c-40d8-985f-02d3f39c948b]8;;\

Started synthetic dataset generation

Output()

🎉 Your synthetic dataset is ready! Use it to consume the generated data. Share it so others can do the same.

latitude  longitude  sex  age  state
0     -1.182548   1.997387    0    2      0
1       0.76447  -1.224311    1    4      0
2      0.795372  -0.777893    0    4      0
3     -1.708918   1.049877    0    2      0
4       0.88762  -0.807637    0    4      0
...         ...        ...  ...  ...    ...
67115  0.809655  -0.783382    0    4      0
67116 -0.976759   1.070693    0    2      0
67117 -0.463864  -0.190423    1    2      0
67118 -0.345393  -0.290523    1    5      0
67119  0.821947   -0.80782    0    4      0

[134240 rows x 5 columns]

Split the generated synthetic data into 20 data sets.

In [9]:
split_sd = np.array_split(sd.data(), 20)

C:\ProgramData\anaconda3\Lib\site-packages\numpy\core\fromnumeric.py:59: FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
  return bound(*args, **kwds)


In [12]:
for i, sX in enumerate(split_sd):
    sX.to_csv("../Data/SK/Synthetic Datasets/mostly_" + str(i) + ".csv", index=False)